content/blog/multitenancy-for-kubernetes-clusters-using-hpe-alletra-6000-and-nimble-storage.md

```diff
@@ -0,0 +1,235 @@
+
+  --remove_folder tenant_name      Remove a folder path from tenant access.
+    --name folder_name             Name of the folder path (pool_name:fqn) to
+                                   be removed (mandatory).
+
+  --passwd                         Change tenant's login password.
+    --tenant name                  Change a specific tenant's login password
+                                   (mandatory).
+```
+
+With these basic create, read, update and delete (CRUD) elements, the storage administrator is now empowered to delegate and confine all the storage resource management to a folder for a Kubernetes administrator to household with.
+
+# The storage administrator's workflow
+
+Assume a new Kubernetes environment is being deployed within an Enterprise. Each cluster needs to be compartmentalized to not consume all the performance and capacity of the array.
+
+First step, create a new restricted folder in the pool of your choice.
+
+```markdown
+folder --create k8s-prod \
+  --iops_limit 75000 --usage_limit 2500000 \
+  --description="Kubernetes Production Cluster"
+```
+
+Next, create a new tenant. 
+
+```markdown
+tenantadmin --add K8sAdminProd --folders default:/k8s-prod
+Enter new password: ********
+Retype new password: ********
+Created User K8sAdminProd
+```
+
+**Hint:** Username must be alphanumeric, cannot start with a number and cannot exceed 32 characters in length. The password policy enforced is derived from the system policy.
+
+At this point, the storage administrator hands over the credentials to the Kubernetes administrator.
+
+It's important to understand that giving the folder name to the Kubernetes administrator is completely optional. This could be useful in situations where it has been determined that a tenant need different performance characteristics for the folders. Like a "gold", "silver" and "bronze" scheme. The Container Storage Provider (CSP) will, by default, pick the folder with the most available capacity for the tenant. The upside by omitting the folder information to the Kubernetes administrator is that the storage administrator has all the power and flexibility to grow storage to new pools for a tenant without the tenant knowing about it. A very popular cloud operational model.
+
+# Apply Kubernetes configuration
+
+Applying the tenant configuration to the Kubernetes cluster is not more difficult than using a standard "administrator" or "poweruser" account on the array.
+
+YAML declarations below are created with:
+
+```markdown
+kubectl create -f-
+< Paste the YAML content >
+Hit CTRL-D on a new line.
+```
+
+Let's create a `Secret` referencing the HPE Alletra 6000 with the tenant credentials.
+
+```yaml
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: hpe-backend
+  namespace: hpe-storage
+stringData:
+  serviceName: alletra6000-csp-svc
+  servicePort: "8080"
+  backend: 192.168.1.30
+  username: K8sAdminProd
+  password: qweqwe123
+```
+
+Next, create a new `StorageClass`.
```

Comment:
Suggestion to specify the storageclass you are creating is marked as the default Storage Class for the K8s cluster:

Next, create a new `StorageClass` and make it the default storage class for our Kubernetes cluster.